dask/blockwise.py

```diff
@@ -244,6 +265,62 @@ def _out_numblocks(self):
         return out_d
 
 
+class BlockwiseGetitem(Blockwise):
+    """
+    Blockwise dedicated to `__getitem__`
+
+    Parameters
+    ----------
+    getitem_key : Scalar or BlockwiseFilter
+    """
+
+    def __init__(
+        self,
+        output,
+        output_indices,
+        dsk,
+        indices,
+        numblocks,
+        concatenate=None,
+        new_axes=None,
+        getitem_key=None,
+    ):
+        super().__init__(
+            output, output_indices, dsk, indices, numblocks, concatenate, new_axes
+        )
+        self.getitem_key = getitem_key
+
+    @property
+    def can_pushdown(self):
+        """Whether this represents a getitem that can be pushed down.
+
+        In the following
+
+        >>> df[df['A'] == 5]
+
+        The outer would be pushed down, while the inner would not.
+        """
+        return isinstance(self.getitem_key, BlockwiseFilter)
+
+
+class BlockwiseFilter(Blockwise):
+    def __init__(
+        self,
+        output,
+        output_indices,
+        dsk,
+        indices,
+        numblocks,
+        concatenate=None,
+        new_axes=None,
+        filter_key=None,
```

Comment:
I might call this something like `filter_expression` or `filter_condition`, since it's a tuple of `(callable, other)`


```
In [28]: df2.dask.layers[df2._name].getitem_key.filter_key
Out[28]: (<function _operator.eq(a, b, /)>, 1)
```